src/Azure.Functions.Cli/Actions/AzureActions/PublishFunctionAppAction.cs

```diff
@@ -243,14 +249,13 @@ public override async Task RunAsync()
                 throw new CliException($"'{FunctionAppName}' app is missing AzureWebJobsStorage app setting. That setting is required for publishing consumption linux apps.");
             }
 
-            if (workerRuntime == WorkerRuntime.dotnetIsolated && functionApp.IsLinux)
+            if (workerRuntime == WorkerRuntime.dotnetIsolated)
             {
-                // For dotnet isolated, as function create options are limited, we update an exisiting App to .NET 5 isolated app,
-                // if the user is trying to deploy that. This is why we need to special case this scenario and set the proper
-                // LinuxFxVersion properties when missing.
-                await DotnetIsolatedLinuxValidation(functionApp);
+                await UpdateDotNetIsolatedFrameworkVersion(functionApp, DotnetFrameworkVersion,
+                    (settings) => AzureHelper.UpdateWebSettings(functionApp, settings, AccessToken, ManagementURL));
             }
-            else if (functionApp.IsLinux && !functionApp.IsDynamic && !string.IsNullOrEmpty(functionApp.LinuxFxVersion))
+
+            if (workerRuntime != WorkerRuntime.dotnetIsolated && functionApp.IsLinux && !functionApp.IsDynamic && !string.IsNullOrEmpty(functionApp.LinuxFxVersion))
```

Comment:
curious, why not `else if (functionApp.IsLinux &&  ...)`?

Reply:
good question... think it was an accident. I'll adjust.